tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,344 @@
+        'releasever': '7',
+        'reposet': 'Red Hat Enterprise Linux 7 Server - Extras (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    },
+    'rhel9_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Content View')
+    return module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    return module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
+
+
+@pytest.fixture
+def verify_target_repo_on_satellite(
+    module_target_sat,
+    function_leapp_cv,
+    module_sca_manifest_org,
+    module_leapp_lce,
+    upgrade_path,
+    function_leapp_ak,
+):
+    """Verify target rhel version repositories has enabled on Satellite Server"""
+    target_rhel_major_ver = upgrade_path['target_version'].split('.')[0]
+    cmd_out = module_target_sat.cli.Repository.list(
+        {
+            'search': f'content_label ~ rhel-{target_rhel_major_ver}',
+            'content-view-id': function_leapp_cv.id,
+            'organization-id': module_sca_manifest_org.id,
+            'lifecycle-environment-id': module_leapp_lce.id,
+        }
```

Comment:
- Whatever you are saying is fine, i.e host (RHEL8.8) will have `target_version` (BaseOS & AppStream) repos listed unless & untill we set release version (using `subscription-manager --set`) to `RHEL9_VER`.
- As per rhel-upgrade document, we have to make sure that correct `target_rhel_major_ver` repos have been added to the correct `CV`, `LCE` on Satellite.

Reply:
I didn't comment on this fixture before because I wanted to get the PR merged ASAP, but I wanted to share my thoughts now. However, since Gaurav opened the discussion here, let me my perspective on it.

I agree with your perspective that if we instruct Satellite to sync these repositories, they should be there. IMHO, testing whether Satellite does its job in terms of repository enablement and sync should be the job of other tests that we have in our test suite. This way, the Leapp & Satellite integration test can focus solely on testing the integration between Leapp and Satellite.

I hope it makes sense. Let me know what you think.

Reply:
yes, both ways are fine, either on satellite where we're enabling it, or check on host which where satellite will put it eventually(so it's better IMO). 
And, the setup part doesn't seem to be a big deal since until Leapp upgrade works, it looks fine to me, and is non-blocking atm